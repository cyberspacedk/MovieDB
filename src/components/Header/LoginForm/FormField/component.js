/* eslint-disable react/prop-types */
import React from 'react';
import { Field } from 'formik';
import styled from 'styled-components';

const FormItem = styled.div`
  position: relative;
  display: flex;
  justify-content: flex-end;
  input {
    margin-bottom: 3px;
  }
  [name='error-message'] {
    position: absolute;
    width: 100%;
    left: -100%;
    top: 0;
    color: #f70f16;
    font-size: 0.9rem;
  }
`;
const FormField = ({ type, name, placeHolder, value, errors, touched }) => (
  <FormItem>
    {touched[name] && errors[name] && (
      <span name="error-message">{errors[name]}</span>
    )}

    <Field
      type={type}
      name={name}
      placeholder={placeHolder}
      value={value[name]}
    />
  </FormItem>
);

export default FormField;
