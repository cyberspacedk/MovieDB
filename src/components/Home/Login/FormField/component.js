import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Input, Form } from 'antd';

const { Item } = Form;

const FormField = ({
  type,
  name,
  icon,
  field,
  form: { touched, errors },
  ...props
}) => (
  <Item
    validateStatus={touched[name] && errors[name] && 'error'}
    help={errors[name]}
  >
    <Input
      type={type}
      {...field}
      {...props}
      prefix={<Icon type={icon} style={{ color: 'rgba(0,0,0,.25)' }} />}
    />
  </Item>
);

FormField.propTypes = {
  field: PropTypes.shape({ name: PropTypes.string }).isRequired,
  form: PropTypes.shape({
    touched: PropTypes.object,
    errors: PropTypes.object,
  }).isRequired,
  type: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default FormField;
