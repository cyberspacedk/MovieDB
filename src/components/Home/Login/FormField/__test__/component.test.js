import React from 'react';
import { shallow } from 'enzyme';
import FormField from '../component';

describe('Component: FormField', () => {
  const props = {
    field: {
      name: 'password',
    },
    form: {
      errors: {},
      touched: {},
    },
  };

  it('should match its snapshot. Field without errors', () => {
    const container = shallow(<FormField {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should match its snapshot. Field with error', () => {
    const nextProps = {
      ...props,
      form: {
        errors: {
          ...props.form.errors,
          password: 'Too short password',
        },
        touched: {
          ...props.form.touched,
          password: 'Required',
        },
      },
    };
    const container = shallow(<FormField {...nextProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should match its snapshot. Field name empty', () => {
    const nextProps = {
      ...props,
      field: {
        name: '',
      },
    };
    const container = shallow(<FormField {...nextProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should match its snapshot. Field name = username', () => {
    const nextProps = {
      ...props,
      field: {
        name: 'username',
      },
    };
    const container = shallow(<FormField {...nextProps} />);
    expect(container).toMatchSnapshot();
  });
});
