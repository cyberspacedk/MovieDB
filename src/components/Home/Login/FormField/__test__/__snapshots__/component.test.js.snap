// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FormField should match its snapshot. Field with error 1`] = `
<FormItem
  hasFeedback={false}
  help="Too short password"
  validateStatus="error"
>
  <Input
    disabled={false}
    name="password"
    prefix={
      <Icon
        style={
          Object {
            "color": "rgba(0,0,0,.25)",
          }
        }
        type="lock"
      />
    }
    type="password"
  />
</FormItem>
`;

exports[`Component: FormField should match its snapshot. Field without errors 1`] = `
<FormItem
  hasFeedback={false}
>
  <Input
    disabled={false}
    name="password"
    prefix={
      <Icon
        style={
          Object {
            "color": "rgba(0,0,0,.25)",
          }
        }
        type="lock"
      />
    }
    type="password"
  />
</FormItem>
`;
